q2_types/per_sample_sequences/_transformer.py

```diff
@@ -26,11 +26,37 @@
 
 
 class PerSampleDNAIterators(dict):
-    pass
+    def __getitem__(self, key):
+        return super().__getitem__(key)()
+
+    def __iter__(self):
+        for sample_id, seqs in self.items():
+            yield sample_id, seqs()
```

Comment:
None of this should be necessary if the dict already stores the generators ready to go